src/protocols/multi_party_ecdsa/gg_2018/party_i.rs

```diff
@@ -210,23 +210,24 @@ impl Keys {
                 ) == bc1_vec[i].com
                     && bc1_vec[i].correct_key_proof.verify(&bc1_vec[i].e).is_ok()
             })
-            .all(|x| x == true);
+            .all(|x| x);
 
         let (vss_scheme, secret_shares) =
             VerifiableSS::share(params.threshold, params.share_count, &self.u_i);
-        match correct_key_correct_decom_all {
-            true => Ok((vss_scheme, secret_shares, self.party_index.clone())),
-            false => Err(InvalidKey),
+        if correct_key_correct_decom_all {
+            Ok((vss_scheme, secret_shares, self.party_index))
+        } else {
+            Err(InvalidKey)
         }
     }
 
     pub fn phase2_verify_vss_construct_keypair_phase3_pok_dlog(
         &self,
         params: &Parameters,
-        y_vec: &Vec<GE>,
-        secret_shares_vec: &Vec<FE>,
-        vss_scheme_vec: &Vec<VerifiableSS>,
-        index: &usize,
+        y_vec: &[GE],
+        secret_shares_vec: &[FE],
+        vss_scheme_vec: &[VerifiableSS],
+        index: usize,
```

Comment:
There are couple of places where you change `&usize` to `usize`. 
why is that better? 
I don't really mind to make this change but it breaks backward compatibility

Reply:
&size is basically a pointer, and its size is even the same as usize
If our pointer is not smaller than the object it points, we should use the object itself.

This is also a clippy warning

Reply:
https://rust-lang.github.io/rust-clippy/current/index.html#trivially_copy_pass_by_ref